src/gt4py/lazy_stencil.py

```diff
@@ -39,6 +39,7 @@ class LazyStencil:
 
     def __init__(self, builder: "StencilBuilder"):
         self.builder = builder
+        self.builder.capture_externals()
```

Comment:
Could this be a call to `self.builder.caching.capture_externals` instead? That would avoid having to define it as a method in the builder as well.